tests/testthat/test-shinytest2-teal_data_module.R

```diff
@@ -0,0 +1,125 @@
+testthat::test_that("e2e: teal_data_module will have a delayed load of datasets", {
+  tdm <- teal_data_module(
+    ui = function(id) {
+      ns <- shiny::NS(id)
+      shiny::actionButton(ns("submit"), label = "Load data")
+    },
+    server = function(id) {
+      shiny::moduleServer(id, function(input, output, session) {
+        shiny::eventReactive(input$submit, {
+          data <- within(
+            teal_data(),
+            {
+              dataset1 <- iris
+              dataset2 <- mtcars
+            }
+          )
+          datanames(data) <- c("dataset1", "dataset2")
+
+          data
+        })
+      })
+    }
+  )
+
+  app <- TealAppDriver$new(
+    data = tdm,
+    modules = example_module(label = "Example Module"),
+    timeout = default_idle_timeout
+  )
+
+  app$click("teal_data_module-submit")
+  app$wait_for_idle()
+  testthat::expect_setequal(app$get_active_filter_vars(), c("dataset1", "dataset2"))
+
+  app$stop()
+})
+
+testthat::test_that("e2e: teal_data_module fails to proceed without input", {
```

Comment:
I like the test. However, I think the desc can be renamed. I believe that we are testing "tdm shows any validation errors" else we're just testing this specific instance of the app.

Reply:
There is a `app$expect_validation_error()` to test this.

Reply:
:+1: Renamed it to your suggestion 016d28f

`expect_validation_error` is being used, what do you mean?

Reply:
True, Sorry I did not see the `expect_validation_error`.